### PR TITLE
Additional external ES changes

### PR DESCRIPTION
**Pull request by caseydavenport**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by tmjd — APPROVED** [review]
LGTM

File: api/v1/tenant_types.go [hunk 1]
```diff
@@ -116,6 +116,20 @@ func (t *Tenant) ElasticMTLS() bool {
...
 
+// MultiTenant returns true if this management cluster is configured to support multiple tenants, and false otherwise.
+func (t *Tenant) MultiTenant() bool {
+	// In order to support multiple tenants, the tenant CR must not be nil, and it must be assigned to a namespace.
+	return t != nil && t.GetNamespace() != ""
+}
+
+// SingleTenant returns true if this management cluster is scoped to a single tenant, and false if this is
+// either a multi-tenant management cluster or a cluster with no tenancy enabled.
+func (t *Tenant) SingleTenant() bool {
```
Comment (Josh-Tigera) [inline api/v1/tenant_types.go:127]: Looks like this is never actually used yet, is it just to provide a means of positively checking for the thing you're most interested in vs having to check for the negation of the thing you're not?

File: pkg/render/logstorage/linseed/linseed_test.go [hunk 2]
```diff
@@ -483,6 +484,89 @@ var _ = Describe("Linseed rendering tests", func() {
...
+			d := rtest.GetResource(resources, DeploymentName, cfg.Namespace, appsv1.GroupName, "v1", "Deployment").(*appsv1.Deployment)
+			envs := d.Spec.Template.Spec.Containers[0].Env
+			Expect(envs).To(ContainElement(corev1.EnvVar{Name: "MANAGEMENT_OPERATOR_NS", Value: "tigera-operator"}))
+			Expect(envs).To(ContainElement(corev1.EnvVar{Name: "LINSEED_EXPECTED_TENANT_ID", Value: cfg.Tenant.Spec.ID}))
+
+			// These are only set for multi-tenant clusters. Make sure they aren't set here.
+			Expect(envs).NotTo(ContainElement(corev1.EnvVar{Name: "BACKEND", Value: "elastic-single-index"}))
+			Expect(envs).NotTo(ContainElement(corev1.EnvVar{Name: "LINSEED_MULTI_CLUSTER_FORWARDING_ENDPOINT", Value: fmt.Sprintf("https://tigera-manager.%s.svc:9443", cfg.Tenant.Namespace)}))
+		})
+	})
```
Comment (Josh-Tigera) [inline pkg/render/logstorage/linseed/linseed_test.go:569]: Maybe a little paranoid but could check that in the single-tenant case we also don't render the multi-tenant env vars.

File: pkg/render/manager_test.go [hunk 3]
```diff
@@ -994,6 +994,33 @@ var _ = Describe("Tigera Secure Manager rendering tests", func() {
...
+					},
+				},
+			})
+			d := rtest.GetResource(tenantAResources, "tigera-manager", render.ManagerNamespace, appsv1.GroupName, "v1", "Deployment").(*appsv1.Deployment)
+			envs := d.Spec.Template.Spec.Containers[2].Env
+			esProxyEnv := d.Spec.Template.Spec.Containers[1].Env
+			Expect(envs).To(ContainElement(corev1.EnvVar{Name: "VOLTRON_TENANT_ID", Value: "tenant-a"}))
+			Expect(esProxyEnv).To(ContainElement(corev1.EnvVar{Name: "VOLTRON_URL", Value: fmt.Sprintf("https://tigera-manager.%s.svc:9443", render.ManagerNamespace)}))
+		})
+	})
```
Comment (Josh-Tigera) [inline pkg/render/manager_test.go:1023]: Same thing here, could verify that we're not rendering the multi-tenant env vars.

File: pkg/render/logstorage/linseed/linseed_test.go (hunk 2 above)
Comment (caseydavenport) [reply pkg/render/logstorage/linseed/linseed_test.go:569]: But you're right the checks can be better - pushing a commit that does it in a more paranoid way.

**Pull request merged**